### PR TITLE
[FR] Remove text restriction

### PR DESCRIPTION
**Pull request by maririos**
I don't intend to merge this until I do a full release, but I wanted to get eyes to make sure my fix was the correct one.
I tried it with the file customer provided and ran all recorded tests

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs [hunk 1]
```diff
@@ -39,13 +39,13 @@ internal FormField(string name, FieldValue_internal fieldValue, IReadOnlyList<Re
             Name = name;
             LabelData = null;
 
-            // Bounding box, page and text are not returned by the service in two scenarios:
```
Comment (maririos) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs:42]: I couldn't find a reason why `text` was required here, so after looking at the output from the service, our existing code, and running recordings, it is safe to remove.

Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs:42]: So there's no scenario in which we'll have no bounding box, no page, but a NON-NULL text, right?

**Review by kinelski — APPROVED** [review]

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs (hunk 1 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs:42]: no bounding box, no page, and yes text.
A user sent us a document where this was true :| first time seeing it too.
See service output:
```
"type": "object",
     "valueObject": {
         "CountryRegion": {
         "type": "countryRegion",
         "text": "ISL",
         "valueCountryRegion": "ISL"
     },
```

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs:42]: We also have:
```
"type": "object",
     "valueObject": {
         "CountryRegion": {
         "type": "countryRegion",
         "valueCountryRegion": "ISL"
     },
```

And both are valid, so I see no point of having that `text` property there

**Review by kristapratico — APPROVED** [review]

**Comment by maririos** [comment]
Change added to https://github.com/Azure/azure-sdk-for-net/pull/21701

**Pull request closed without merging**